src/foam/nanos/notification/email/SMTPEmailService.js

```diff
@@ -293,6 +316,57 @@ if ( getAuthenticate() ) {
 } else {
   session_ = Session.getInstance(props);
 }`
+    },
+    {
+      name: 'finalizeEmailConfig',
+      javaReturns: 'foam.nanos.notification.email.EmailMessage',
+      args: [
+        {
+          name: 'x',
+          javaType: 'foam.core.X'
+        },
+        {
+          name: 'emailMessage',
+          javaType: 'final foam.nanos.notification.email.EmailMessage'
+        }
+      ],
+      javaCode:
+        `
+foam.nanos.session.Session session = x.get(foam.nanos.session.Session.class);
+
+DAO userDAO         = (DAO) x.get("localUserDAO");
+DAO groupDAO        = (DAO) x.get("groupDAO");
+
+User user           = (User) userDAO.find(session.getUserId());
+Group group         = (Group) groupDAO.find(user.getGroup());
+AppConfig appConfig = group.getAppConfig(x);
+
+String domain = appConfig.getUrl().contains("ablii") ? "ablii.com" : "nanopay.net";
```

Comment:
Why are explicitly testing for ablii?

Reply:
Just an easier way to get the domain from URL. If URL contains "ablii", then the domain is "ablii", otherwise the domain is "nanopay.net".

Reply:
Yes, but now we need a test for every url we may deploy under, this does not scale.
Consider this quote from a recent email:  `...  we use a different domain to send Marketing emails, so we don't affect our domain rating. For example, we should be using hello@e.ablii.com or hello@email.ablii.com, rather than ablii.com. Same thing goes for nanopay.net.`

Since the outbound domain may not match the request url, simply use, in order:

1. email parameters
2. group parameters
3. smtpemail parameters